gribscan/gribscan.py

```diff
@@ -348,11 +348,14 @@ def write_index(gribfile, idxfile=None, outdir=None, force=False):
     # collapse the "/./" notation used to denote subtrees.
     gen = scan_gribfile(open(p, "rb"), filename=gribfile)
 
-    with open(idxfile, "w" if force else "x") as output_file:
+    tempfile = idxfile.with_suffix(".index.partial")
+    with open(tempfile, "w" if force else "x") as output_file:
         for record in gen:
             json.dump(record, output_file)
             output_file.write("\n")
 
+    tempfile.rename(idxfile)
```

Comment:
Likely the logic behind `force` doesn't work in this case anymore. I'm not yet sure how we get exactly the same behaviour (i.e. not replacing exisiting files without race conditions). But maybe the following would be close enough:

```suggestion
    with open(tempfile, "w") as output_file:
        for record in gen:
            json.dump(record, output_file)
            output_file.write("\n")

    if force or not idxfile.exists():
        tempfile.rename(idxfile)
```

I.e. we'd always overwrite the tempfile (because it's old and broken if it's still there), but we would only overwrite `idxfile` if it's not yet there, or we want to force it. This one will theoretically add a tiny race condition (where `idxfile` would be created after checking it's existence, but before overwriting), but I assume in our use cases this would be fine (and we also would likely have a similar issue with the partial files in this case).